File: Golf.js/schemas/teebox.js
var teeboxSchema = { 
    "type": "object",
        "properties": {
            "color": {
                "type": "string",
                "pattern": "^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$",
                "format": "color"
            },
            "name": {
                "type": "string"
            },
            "distance": {
                "type": "number",
                "minimum": 1
            },
            "par": {
                "type": "number",
                "minimum": 1
            },
            "courseRating": {
                "type": "number",
                "minimum": 1
            },
            "slopeRating": {
                "type": "number",
                "minimum": 1
            },
            "holes": {
                "type": ["array", "null"],
                "items": {
                    "type": "object",
                    "properties": {
                        "holeId": {
                            "type": "string"
                        },
                        "number": {
                            "type": "number",
                            "minimum": 1
                        },
                        "distance": {
                            "type": "number",
                            "minimum": 1
                        },
                        "par": {
                            "type": "number",
                            "minimum": 3,
                            "maximum": 5
                        },
                        "hcp": {
                            "type": "number",
                            "minimum": 1
                        }
                    },
                    "required": ["number", "distance", "par", "hcp"]
                }
            },
            "required": ["color", "name", "distance", "par", "courseRating", "slopeRating"]
        }
};

module.exports = teeboxSchema;

File: Golf.Models/TeeBox.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace Golf.Tournament.Models
{
    public class TeeBox
    {
        public TeeBox()
        {
            Holes = new CourseHoles();
            Color = new Color("#ffffff");
        }

        [JsonProperty("_id")]
        public string Id { get; set; }

        [JsonProperty("color")]
        [Required]
        public Color Color { get; set; }

        [JsonProperty("name")]
        [Required]
        public string Name { get; set; }

        [JsonProperty("distance")]
        [Required]
        public int Distance { get; set; }

        [JsonProperty("par")]
        [Required]
        public int Par { get; set; }

        [JsonProperty("courseRating")]
        [Required]
        public float CourseRating { get; set; }

        [JsonProperty("slopeRating")]
        [Required]
        public float SlopeRating { get; set; }
        
        private CourseHoles holes = new CourseHoles();

        [JsonProperty("holes")]
        public CourseHoles Holes
        {
            get
            {
                return holes;
            }
            set
            {
                holes = value ?? new CourseHoles();
            }
        }
    }
}